apps/previewgenerator.sh

```diff
@@ -16,67 +16,256 @@ debug_mode
 # Check if root
 root_check
 
-msg_box "Please note that this will put your server at risk as it will install a package called 'imagick' which is known to have several flaws.
-
-You can check this issue to understand why: https://github.com/nextcloud/vm/issues/743
+# PHP 7.x is needed
+if is_this_installed php5.6-common
+then
+    msg_box "At least PHP 7.X is supported, please upgrade your PHP version: https://shop.hanssonit.se/product/upgrade-php-version-including-dependencies/"
+    exit
+elif is_this_installed php5.5-common
+then
+    msg_box "At least PHP 7.X is supported, please upgrade your PHP version: https://shop.hanssonit.se/product/upgrade-php-version-including-dependencies/"
+    exit
+fi
 
-You can choose to cancel installing this in the next step."
-if [[ "no" == $(ask_yes_or_no "Do you still want to continue?") ]]
+# Encryption may not be enabled
+if [ -n "$(occ_command app:list | grep encryption | awk '{print $3}')" ]
 then
```

Comment:
Is this working for you?
I haven't tested it myself.
And what about end to end encryption?
Shouldn't that get checked here, too?

Reply:
I guess my mistake. Should work with both, since they both have "encryption" in their name.